x-pack/plugins/security_solution/public/common/components/super_date_picker/selectors.ts

```diff
@@ -8,19 +8,30 @@
 import { isEmpty } from 'lodash';
 import { createSelector } from 'reselect';
 import type { State } from '../../store';
-import type { InputsModelId } from '../../store/inputs/constants';
-import type { Policy, InputsRange, TimeRange, GlobalQuery } from '../../store/inputs/model';
+import type {
+  GlobalQuery,
+  GlobalKqlQuery,
+  InputsRange,
+  InputsRangeTimeOnly,
+  Policy,
+  TimeRange,
+} from '../../store/inputs/model';
 
-export const getPolicy = (inputState: InputsRange): Policy => inputState.policy;
+export const getPolicy = (inputState: InputsRange | InputsRangeTimeOnly): Policy =>
+  inputState.policy;
 
-export const getTimerange = (inputState: InputsRange): TimeRange => inputState.timerange;
+export const getTimerange = (inputState: InputsRange | InputsRangeTimeOnly): TimeRange =>
+  inputState.timerange;
 
 export const getQueries = (inputState: InputsRange): GlobalQuery[] => inputState.queries;
 
-export const getGlobalQueries = (state: State, id: InputsModelId): GlobalQuery[] => {
+export const getGlobalQueries = (state: State, id: 'global' | 'timeline'): GlobalQuery[] => {
   const inputsRange = state.inputs[id];
   return !isEmpty(inputsRange.linkTo)
     ? inputsRange.linkTo.reduce<GlobalQuery[]>((acc, linkToId) => {
+        if (linkToId === 'socTrends') {
+          return acc;
+        }
```

Comment:
Same here, we could use the `isQueryInput` to check if it needs the `queries` or not

Reply:
I can do this once the feature flag is lifted, but right now it is possible for `state.inputs[linkToId]` to be undefined, so checking explicitly for `socTrends` is cleaner
<img width="1054" alt="Screen Shot 2022-08-30 at 8 36 27 AM" src="https://user-images.githubusercontent.com/6935300/187465829-ee9e0825-c6bd-4756-83b0-3b884b3a098c.png">